dex/src/main/scala/com/wavesplatform/dex/tool/WaitOffsetTool.scala

```diff
@@ -0,0 +1,76 @@
+package com.wavesplatform.dex.tool
+
+import akka.actor.Scheduler
+import com.wavesplatform.dex.domain.utils.ScorexLogging
+import com.wavesplatform.dex.queue.ValidatedCommandWithMeta.Offset
+import com.wavesplatform.dex.settings.MatcherSettings
+
+import java.util.concurrent.TimeoutException
+import scala.concurrent.{ExecutionContext, Future, Promise}
+import scala.concurrent.duration.Deadline
+import scala.util.{Failure, Success}
+import scala.concurrent.duration._
+
+trait WaitOffsetTool extends ScorexLogging {
+
+  case class TimeCheckTag(currentOffset: Offset, time: Long)
+
+  def waitOffsetReached(
+    getLastOffset: => Future[Offset],
+    getLastProcessedOffset: => Offset,
+    startingLastOffset: Offset,
+    deadline: Deadline,
+    settings: MatcherSettings,
```

Comment:
we have exception that uses some configs for message. I think it may be useful to see this setting in logs if something went wrong 
```
p.failure(new TimeoutException(s"Can't process all events in ${settings.startEventsProcessingTimeout.toMinutes} minutes"))
```

Reply:
we can add this setting to WaitingOffsetToolSettings as well (as i understood it's the only param)

Reply:
Agree, we can write in settings:
```hocon
waiting-offset-tool {
  foo = ${waves.dex.bar}
}
```
and then use this "foo".